### PR TITLE
Disaster Scans

### LINKED ISSUE #16596: Disaster Scans Website Changes
### Source information

Disaster Scans 1.3.30

### Source language

English

### Steps to reproduce

They changed their website so the source is dead

### Expected behavior

Load new chapter 

### Actual behavior

Error 404

### Tachiyomi version

0.13.5

### Android version

Android 11

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.14.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by AwkwardPeak7**
closes #16596

Stuff that may need to be addressed in the future:
- No pagination
- Client side search
- Better way to get slugs

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+@Serializable
+data class ApiSearchComic(
+    val id: String,
+    val ComicTitle: String,
+    val CoverImage: String,
+) {
+    fun toSManga(cdnUrl: String) = SManga.create().apply {
+        title = ComicTitle
+        thumbnail_url = "$cdnUrl$CoverImage#thumbnail"
+        url = "/comics/$id-${ComicTitle.titleToSlug()}"
```
Comment (AntsyLich) [inline src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt:18]: Just keep the id if that works please.

File: src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    val CoverImage: String,
+    val Status: String,
+    val Genres: String,
+    val Author: String,
+    val Artist: String,
+) {
+    fun toSManga(json: Json, cdnUrl: String) = SManga.create().apply {
+        title = ComicTitle
+        thumbnail_url = "$cdnUrl$CoverImage#thumbnail"
+        url = "/comics/$id-${ComicTitle.titleToSlug()}"
```
Comment (AntsyLich) [inline src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt:36]: Same here

File: src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+@Serializable
+data class ApiChapter(
+    val chapterID: Int,
+    val chapterNumber: String,
+    val ChapterName: String,
+    val chapterDate: String,
+) {
+    fun toSChapter(mangaUrl: String) = SChapter.create().apply {
+        url = "$mangaUrl/$chapterID-chapter-$chapterNumber"
```
Comment (AntsyLich) [inline src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt:53]: Also here

File: src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt (hunk 1 above)
Comment (AwkwardPeak7) [reply src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt:18]: `id` alone doesn't work

Comment (AwkwardPeak7) [reply src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScansDto.kt:18]: added slug based search to bypass potential problems with `titleToSlug`

**Pull request merged**